### PR TITLE
A numeric value for value_box()'s title/value is now wrapped in a header tag (just like a string is)

### LINKED ISSUE #512: Better handling of numeric value in `value_box()`
Currently doesn't get wrapped properly in a "header container" (e.g., `value_box("Foo", 4)`)

### PR DESCRIPTION
**Pull request by cpsievert**
Closes #512

### CODE REVIEW

File: R/value-box.R [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #' even a [shiny::plotOutput()]).
 #'
 #' @param title,value a [htmltools::tag()] child to display above `value`. If a
-#'   string a provided, it's automatically wrapped in a header tag.
+#'   string (or number) a provided, it's automatically wrapped in a header tag.
```
Comment (gadenbuie) [inline R/value-box.R:9]: wdyt about this?

```r
#' @param title,value A string, number, or [htmltools::tag()] to display as the
#'   title or value of the value box. The `title` appears above the `value`.
```

Unless you want to get into the default classes added to each item, I think

> automatically wrapped in a header tag

is slightly misleading. I think it's probably better to put string/number front and center.

**Review by gadenbuie — APPROVED** [review]
Looks good to me! One small comment/suggestion for the documentation.

Also I noticed that the user will get an error if they pass a vector of values to `title` or `value`, but it's not very informative.

``` r
value_box(c("A", "B"), 1:2)
#> Error in writeImpl(text): Text to be written must be a length-one character vector
```

File: R/value-box.R [hunk 2]
```diff
@@ -47,10 +47,10 @@ value_box <- function(title, value, ..., showcase = NULL, showcase_layout = show
   attribs <- args[nzchar(argnames)]
   children <- args[!nzchar(argnames)]
 
-  if (rlang::is_bare_character(title)) {
+  if (rlang::is_bare_character(title) || rlang::is_bare_numeric(value)) {
```
Comment (gadenbuie) [inline R/value-box.R:50]:
```suggestion
  if (rlang::is_bare_character(title) || rlang::is_bare_numeric(title)) {
```

File: R/value-box.R (hunk 1 above)
Comment (cpsievert) [reply R/value-box.R:9]: Yes, way better, thanks

File: R/value-box.R (hunk 2 above)
Comment (cpsievert) [reply R/value-box.R:50]: Nice catch!!!

File: R/value-box.R (hunk 1 above)
Comment (cpsievert) [inline R/value-box.R:9]:
```suggestion
#' @param title,value A string, number, or [htmltools::tag()] child to display as the
#'   title or value of the value box. The `title` appears above the `value`.
```

**Pull request merged**